packages/htmlbars-runtime/lib/hooks.js

```diff
@@ -258,6 +259,11 @@ function yieldItem(template, env, parentScope, morph, renderState, visitor) {
       handledMorphs[foundMorph.key] = foundMorph;
       yieldTemplate(template, env, parentScope, foundMorph, renderState, visitor)(blockArguments, self);
     } else {
+      if(morph.childNodes){
+         visitChildren(morph.childNodes, function (node) {
+           clearMorph(node, env, true);
+         });
+       }
```

Comment:
Can someone explain the logic of this change? I don't see how clearing all the morphs when adding a new item is the right thing to do.